src/corelibs/U2Core/src/util/FileAndDirectoryUtils.cpp

```diff
@@ -142,4 +143,13 @@ QString FileAndDirectoryUtils::getAbsolutePath(const QString &filePath) {
     return QFileInfo(result).absoluteFilePath();
 }
 
+bool FileAndDirectoryUtils::isDirectoryWritable(const QString &path) {
+    if (!QFileInfo(path).isDir()) {
+        return false;
+    }
+
+    QTemporaryFile tmp(path + "/XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX");
```

Comment:
Are you sure that the file is created on the file system with no call to file.open() ?